### PR TITLE
Make trailing comma logic more consise

### PR DESCRIPTION
**Pull request by RedGuy12**
### Description

Simplified the logic for adding trailing commas. Turned out it was actually a bit simpler than I thought.
See https://github.com/psf/black/pull/4164#discussion_r1466689991

### Checklist - did you ...

- [-] Add an entry in `CHANGES.md` if necessary?
- [-] Add / update tests if necessary?
- [-] Add new / update outdated documentation?

### CODE REVIEW

File: src/black/linegen.py [hunk 1]
```diff
@@ -1050,22 +1051,9 @@ def bracket_split_build_line(
...
-                    for node in (
-                        leaves[0].parent,
-                        getattr(leaves[0].parent, "parent", None),
-                    )
-                    if isinstance(node, Node) and isinstance(node.prev_sibling, Leaf)
-                )
-                # Except the false negatives above for PEP 604 unions where we
-                # can't add the comma.
+                # Don't add one inside parenthesized return annotations
+                and not get_annotation_type(leaves[0]) == "return"
```
Comment (JelleZijlstra) [inline src/black/linegen.py:1055]:
```suggestion
                and get_annotation_type(leaves[0]) != "return"
```

File: src/black/nodes.py [hunk 2]
```diff
@@ -951,16 +962,21 @@ def is_number_token(nl: NL) -> TypeGuard[Leaf]:
     return nl.type == token.NUMBER
 
 
-def is_part_of_annotation(leaf: Leaf) -> bool:
-    """Returns whether this leaf is part of type annotations."""
+def get_annotation_type(leaf: Leaf) -> Optional[Literal["return", "param"]]:
```
Comment (JelleZijlstra) [inline src/black/nodes.py:965]:
```suggestion
def get_annotation_type(leaf: Leaf) -> Literal["return", "param", None]:
```

**Review by JelleZijlstra — COMMENTED** [review]
Thanks, two nits

**Review by hauntsaninja — APPROVED** [review]

**Review by hauntsaninja — APPROVED** [review]
Thank you again! :-)

**Pull request merged**